設定画面のアイテムの実装

sanadryujiwa opened the pull request:
## 目的
設定画面のアイテムの実装

## 方針
- 設定フラグメント内のRecyclerViewでアイテム別の処理を行うプログラムを書く
- 通信処理の実装を行う

## 実装
- 設定フラグメントからアダプターを呼び出すとき、アイテム別の処理を行うプログラムを渡す
- 設計したAPIをまとめたHackMDをもとに、モデル、インターフェース、ビューモデルを記述した

## テスト
- 設定フラグメントでRecyclerViewのアイテムをタップし、「ご意見・ご要望」→Requestの画面、「不具合報告」→FeedBackの画面に飛べることを確認する
- それぞれの画面でテキスト入力、送信が行えることを確認する

sanadryujiwa committed:
設定フラグメントの「ご意見・ご要望」の通信処理の記述

sanadryujiwa committed:
通信処理の修正

sanadryujiwa committed:
「ご意見・ご要望」、「不具合報告」の通信処理実装

sanadryujiwa committed:
レイアウトのリソース値の書き出し

sanadryujiwa committed:
RecyclerViewのアイテム別の処理の実装

sanadryujiwa committed:
data classの記述場所の変更とレイアウトの編集

sanadryujiwa committed:
レイアウト画面の作成

sanadryujiwa committed:
修正（パッケージの追加）

sanadryujiwa committed:
エラーの修正

sanadryujiwa committed:
カーソルの処理の修正

Ryuki0222 commented on app/src/main/java/com/example/triplan/api/repository/SettingRepository.kt:
```diff
@@ -0,0 +1,80 @@
+import retrofit2.Callback
+import retrofit2.Response
+
+class SettingRepository {
```
CommonRepositoryがあるのでそこにこれらの処理を記述してください。

Ryuki0222 commented on app/src/main/java/com/example/triplan/api/service/SettingService.kt:
```diff
@@ -0,0 +1,19 @@
+import retrofit2.http.Headers
+import retrofit2.http.POST
+
+interface SettingService {
```
これもCommonServiceにまとめてください

Ryuki0222 commented on .idea/codeStyles/Project.xml:
```diff
@@ -1,5 +1,8 @@
   <code_scheme name="Project" version="173">
+    <AndroidXmlCodeStyleSettings>
+      <option name="ARRANGEMENT_SETTINGS_MIGRATED_TO_191" value="true" />
+    </AndroidXmlCodeStyleSettings>
```
これけして

Ryuki0222 commented on app/src/main/java/com/example/triplan/api/model/Body/SettingBody.kt:
```diff
@@ -0,0 +1,16 @@
+            return SettingBody(text)
+        }
+    }
+}
```
これ、FeedBackBodyとRequestBodyに分けてもらっていい？

Ryuki0222 commented on app/src/main/java/com/example/triplan/api/model/Json/SettingJson.kt:
```diff
@@ -0,0 +1,8 @@
+data class SettingJson (
+    @Json(name = "done")
+    val done: Boolean
+)
```
これも分けてくだせぇ

Ryuki0222 commented on app/src/main/java/com/example/triplan/ui/setting_contact/SettingContactActivity.kt:
```diff
@@ -0,0 +1,74 @@
+import com.example.triplan.model.Setting
+import kotlinx.android.synthetic.main.activity_setting_contact.*
+
+class SettingContactActivity : AppCompatActivity() {
```
class SettingRequestActivity

Ryuki0222 commented on app/src/main/java/com/example/triplan/ui/setting_contact/SettingContactViewModel.kt:
```diff
@@ -0,0 +1,28 @@
+import com.example.triplan.api.repository.SettingRepository
+import com.example.triplan.model.Setting
+
+class SettingContactViewModel: ViewModel() {
```
SettingRequestViewModel

Ryuki0222 commented on app/src/main/java/com/example/triplan/ui/setting/SettingRecyclerViewAdapter.kt:
```diff
@@ -1,27 +1,46 @@
 
 class SettingRecyclerViewAdapter(
-    private val settingLists: List<String>
+    private val settingLists: List<SettingElement>
```
```
private val settingList: List<SettingElement>,
private val itemOnClickListener: ((View, SettingElement) -> Unit)
```

Ryuki0222 commented on app/src/main/java/com/example/triplan/ui/setting/SettingRecyclerViewAdapter.kt:
```diff
@@ -1,27 +1,46 @@
-        holder.itemView.settingText.text = settingLists.getSafety(position)
+        holder.itemView.settingText.text = settingLists.getSafety(position)?.name
+        holder.itemView.setOnClickListener {
+            listener.onClick(it, settingLists[position])
```
```
val settingElement = settingList.getSafety(position)
holder.itemView.setOnClickListener { view ->
  settingList.getSafety(position)?.let {
    itemOnClickListener.invoke(view, it)
  }
}
```

Ryuki0222 reviewed (commented):
@sanadryujiwa 
コメントした

Ryuki0222 commented on app/src/main/java/com/example/triplan/ui/setting/SettingFragment.kt:
```diff
@@ -20,9 +24,32 @@ class SettingFragment: Fragment() {
+            SettingElement(SettingElementId.FEEDBACK, "不具合報告"),
+            SettingElement(SettingElementId.TERMS_OF_SERVICE, "利用規約")
+        )
         val settingRecyclerViewAdapter = SettingRecyclerViewAdapter(settingLists)
```
val settingRecyclerViewAdapter = SettingRecyclerViewAdapter(settingLists, {
    // ここにクリック処理をする
  }
)

Ryuki0222 commented on app/src/main/java/com/example/triplan/ui/setting_contact/SettingContactActivity.kt:
```diff
@@ -0,0 +1,74 @@
+            val text = settingContactText.text.toString().trim()
+            if (text.isEmpty()) {
+                Toast.makeText(this, "文章を入力してください", Toast.LENGTH_SHORT).show()
+                return@setOnClickListener
```
```
view.isEnable = true
return@setOnClickListener
```

Ryuki0222 reviewed (commented):
追加コメントしたー！

sanadryujiwa committed:
レヴューをもとに修正

- 命名の変更　contact → request
- SettingフラグメントのRecyclerViewのアイテム別の処理の変更
- SettingをRequestとFeedBackに分ける（Body,Json）
- SettingをCommonにまとめる（Repository, Service）

sanadryujiwa committed:
クエリにトークンを付与

Ryuki0222 commented on app/src/main/java/com/example/triplan/api/model/Body/FeedBackBody.kt:
```diff
@@ -0,0 +1,16 @@
+            text: String
+        ): FeedBackBody {
+            return FeedBackBody(text)
+        }
```
setBodyいらんくね？

Ryuki0222 commented on app/src/main/java/com/example/triplan/api/model/Body/RequestBody.kt:
```diff
@@ -0,0 +1,16 @@
+            text: String
+        ): RequestBody {
+            return RequestBody(text)
+        }
```
setBodyいらんくね？

Ryuki0222 commented on app/src/main/java/com/example/triplan/api/service/CommonService.kt:
```diff
@@ -1,15 +1,23 @@
 import retrofit2.Call
-import retrofit2.http.GET
-import retrofit2.http.Path
+import retrofit2.http.*
```
*になってるから直して

Ryuki0222 reviewed (commented):
@sanadryujiwa 
ちょいと、レビューしました

sanadryujiwa committed:
レビューをもとに修正
